internal/wal_restore_handler.go

```diff
@@ -0,0 +1,139 @@
+package internal
+
+import (
+	"errors"
+	"github.com/spf13/viper"
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/utility"
+	"os"
+	"path/filepath"
+	"sort"
+)
+
+// Types for sorting
+type TimelineSlice []uint32
+
+func (p TimelineSlice) Len() int { return len(p) }
+
+// Because we need sorted slice in descending order
+func (p TimelineSlice) Less(i, j int) bool { return p[i] > p[j] }
+func (p TimelineSlice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+type WalSegmentNumbers []WalSegmentNo
+
+func (p WalSegmentNumbers) Len() int { return len(p) }
+
+// Because we need sorted slice in descending order
+func (p WalSegmentNumbers) Less(i, j int) bool { return p[i] > p[j] }
+func (p WalSegmentNumbers) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+// HandleWALRestore is invoked to perform wal-g wal-restore
+func HandleWALRestore(externalFolder storage.Folder, localFolder storage.Folder) {
+	externalHistoryRecs, err := getTimeLineHistoryRecords(1, externalFolder)
+	tracelog.ErrorLogger.FatalfOnError("Failed to get the external WAL history records %v\n", err)
+	localHistoryRecs, err := getTimeLineHistoryRecords(1, localFolder)
```

Comment:
Actually I am not sure if we can trust the history record file to get the current timeline and LSN, Postgres stores this information in the `pg_control` file and I think it should be preffered to get the precise information.